test/test_torch.py

```diff
@@ -2632,6 +2632,16 @@ def _test_scatter_base(self, cast, method, is_scalar=False, test_bounds=True):
             with self.assertRaises(RuntimeError):
                 getattr(base.clone(), method)(dim, idx, src)
 
+        # should throw an error when self.dtype != src.dtype.
+        # we ignore the case when src is Scalar, as it gets
+        # cast via src.to<scalar_t>.
+        if not is_scalar:
+            with self.assertRaises(RuntimeError):
```

Comment:
nit: can you use assertRaisesRegexp, to make sure you are catching an error you are expecting to catch and not something else? 
Let me know when you resolve the conflicts and it is ready for merging.